mautic/mautic
# Send email to user preview fix

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ Yes]
| New feature/enhancement? (use the a.x branch)      | [No ]
| Deprecations?                          | [ NA]
| BC breaks? (use the c.x branch)        | [ NA]
| Automated tests included?              | [ Yes] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 



#### Description:

When setting up a point trigger action to send an email to user, the preview button showed a 404 error on click and the edit email button led to the email list view page instead of the email edit page. Additionally, while editing an existing trigger with the same event, the preview and edit email buttons were disabled.

The wrong link redirection was a result of 'undefined' being passed as the email ID. This was due to the fact that the email id was being read from [email], where as the actual value was stored in [useremail][email] key. This was fixed by fixing 'update-select' attribute in formTypeOptions while building the trigger.

Similarly, the buttons were disabled based on !isset($options['data']['email']) check. Since this [email] key wasn't available in this case, the buttons were disabled anyway. Fixed that by adding another check for the existing key.



#### Steps to test this PR:



1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs here)
2. Go to Points -> Manage Triggers
3. Add new trigger and add event "Send an email to user"
4. Add an email and click Preview
5. A new window with the correct preview should be seen
6. Go back and click Edit Email
7. The edit email page for the selected email should open up in a new window
8. Save this event and go back to edit it
9. The preview and edit email buttons should be enabled
10. Remove the email selection
11. Now the buttons should be disabled





<a href="https://gitpod.io/#https://github.com/mautic/mautic/pull/11449"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>
